### ISSUE TITLE
Clicking on search box toggles should not run search on homepage

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#34718]
On the homepage, clicking on the search box toggles (case sensitivity, pattern type) should not execute the search. The search should only be executed once the search button is clicked (or enter is pressed while focusing on the query).

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.

**Comment by camdencheek** [comment]
Just confirmed this is still an issue

**Comment by fkling** [comment]
I don't think we should do this atm. This can lead to the following (annoying IMO) behavior:

1. User types query
2. User clicks e.g. case sensitive button
3. User has to click back into the search input
4. User presses enter to submit query

That's two steps more than what you'd normally have to do. And doing (3) isn't obvious. It's not as bad in the old search input because it has a dedicated submit button, so that' only 3 steps and they are more obvious.
